plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx

```diff
@@ -105,15 +113,63 @@ export const KubernetesContent = ({ entity }: KubernetesContentProps) => {
   >(undefined);
   const [error, setError] = useState<string | undefined>(undefined);
 
+  const configApi = useApi(configApiRef);
+  const clusters: Config[] = configApi.getConfigArray('kubernetes.clusters');
+  const allAuthProviders: string[] = clusters.map(c =>
+    c.getString('authProvider'),
+  );
+  const authProviders: string[] = [...new Set(allAuthProviders)];
+
+  // Create mapping for supported auth providers to their API refs
+  const authProviderApiMapping: Map<string, ApiRef<OAuthApi>> = new Map<
+    string,
+    ApiRef<OAuthApi>
+  >();
+  authProviderApiMapping.set('google', googleAuthApiRef);
+
+  // These and their authProviders have to be instantiated outside of the useEffect hook
+  const kubernetesAuthProviderMap: Map<
+    string,
+    KubernetesAuthProvider
+  > = new Map<string, KubernetesAuthProvider>();
+  // TODO: Query Kubernetes backend to get info on which auth providers are in use
+  // For each auth type, create instance of KubernetesAuthProvider
+  for (const authProviderStr of authProviders) {
+    let api;
+    if (authProviderApiMapping.has(authProviderStr)) {
+      api = useApi(authProviderApiMapping.get(authProviderStr)!);
+    }
+    const kubernetesAuthProvider: KubernetesAuthProvider = KubernetesAuthProviderGenerator.getKubernetesAuthProviderInstance(
+      authProviderStr,
+      api,
+    );
+    kubernetesAuthProviderMap.set(authProviderStr, kubernetesAuthProvider);
+  }
+
   useEffect(() => {
-    kubernetesApi
-      .getObjectsByServiceId(entity.metadata.name)
-      .then(result => {
-        setKubernetesObjects(result);
-      })
-      .catch(e => {
-        setError(e.message);
-      });
+    (async () => {
+      // For each auth type, invoke decorateRequestBodyForAuth on corresponding KubernetesAuthProvider
+      let requestBody: AuthRequestBody = {};
+      for (const authProviderStr of authProviders) {
+        const kubernetesAuthProvider: KubernetesAuthProvider = kubernetesAuthProviderMap.get(
+          authProviderStr,
+        )!;
+        // Multiple asyncs done sequentially instead of all at once to prevent same requestBody from being modified simultaneously
+        requestBody = await kubernetesAuthProvider.decorateRequestBodyForAuth(
+          requestBody,
+        );
+      }
+
+      // TODO: Add validation on contents/format of requestBody
+      kubernetesApi
+        .getObjectsByServiceId(entity.metadata.name, requestBody)
+        .then(result => {
+          setKubernetesObjects(result);
+        })
+        .catch(e => {
+          setError(e.message);
+        });
+    })();
```

Comment:
It seems I need to add `authProviders` and `kubernetesAuthProviderMap` to this dependency list for `useEffect`. However, doing so seems to cause the Kubernetes page to re-render the KubernetesContent component (and hence make Kubernetes resource fetch requests in an endless loop). Via `console.log` usage that logged all 4 of these dependencies with each render/invocation of the `useEffect` hook, it didn't seem like there were any differences in the values of any of these 4 variables, so I'm not sure why the re-rendering is occurring.

Will need to look more into this possibly as the infinite looping is inefficient/a strain on the backend.

Reply:
Added into the comment a comment to have the linter ignore the need to have `authProviders` in the dependency array (I saw the `eslint-disable react-hooks/exhaustive-deps` being used in a few other plugins).